### ISSUE TITLE
Pods stuck in terminating state after AMI amazon-eks-node-1.15-v20200312

### ISSUE THREAD

**Issue by harshal-shah** [awslabs/amazon-eks-ami#443]
**What happened**:
Since upgrade to AMI `amazon-eks-node-1.15-v20200312`, we see a lot of pods stuck in `Terminating` state

**What you expected to happen**:
When a deployment is upgraded, the old pods should get deleted and then removed.

**How to reproduce it (as minimally and precisely as possible)**:
This is what we see in our case.

The pod (and many others in a similar situation) `process-orchestrator-consumer-65b9c775b8-fn888` is stuck in terminating state. 

ssh into the node that contains the pod: 

~~~
docker ps -a | grep process-orchestrator-consumer-65b9c775b8-fn888
77a78ddfc5cc        0f20ce6b2efa                                                                       "istio-iptables -p 1…"   19 hours ago         Up 19 hours                                          k8s_istio-init_process-orchestrator-consumer-65b9c775b8-fn888_consumer-core_fe17a231-c76b-43f2-8e3a-3c19f52b6206_147
f6c76c4b1421        602401143452.dkr.ecr.eu-west-1.amazonaws.com/eks/pause-amd64:3.1                   "/pause"                 3 days ago           Exited (0) 8 hours ago                               k8s_POD_process-orchestrator-consumer-65b9c775b8-fn888_consumer-core_fe17a231-c76b-43f2-8e3a-3c19f52b6206_0
~~~

Inspect the container which is being shown as "Up 19 hours" (which ideally should have been stopped and rm'ed)
~~~
docker inspect 77a78ddfc5cc | grep Pid
            "Pid": 29682,
~~~

Check for this PID
~~~
ps -ef | grep 29682
root     26297 22185  0 16:10 pts/0    00:00:00 grep --color=auto 29682
~~~

So the process behind the "Up" container is killed but docker does not know about it and hence it never reports back to kubelet that the pod is terminated. So kubelet never reports back to API server and that's why the pod keeps showing in "Terminating" state

**Anything else we need to know?**:
So far this seems like a docker issue. 
Restarting docker daemon on the node fixes the problem

~~~
$service docker restart
Redirecting to /bin/systemctl restart docker.service
$docker ps -a | grep 77a78ddfc5cc
$docker ps -a | grep 77a78ddfc5cc
~~~

**Environment**:
- AWS Region: eu-west-1
- Instance Type(s): c5.xlarge / c5.2xlarge / t3.medium  
- EKS Platform version (use `aws eks describe-cluster --name <name> --query cluster.platformVersion`): eks.1
- Kubernetes version (use `aws eks describe-cluster --name <name> --query cluster.version`): 1.15
- AMI Version: amazon-eks-node-1.15-v20200312
- Kernel (e.g. `uname -a`): `Linux ip-172-23-76-247.eu-west-1.compute.internal 4.14.171-136.231.amzn2.x86_64 #1 SMP Thu Feb 27 20:22:48 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux`
- Release information (run `cat /etc/eks/release` on a node): 
~~~
BASE_AMI_ID="ami-0e50648bfc0a7d87a"
BUILD_TIME="Thu Mar 12 23:03:39 UTC 2020"
BUILD_KERNEL="4.14.171-136.231.amzn2.x86_64"
ARCH="x86_64"
~~~

**Comment by harshal-shah** [comment]
This problem seems to happen due to the following issue https://github.com/docker/for-linux/issues/555 
We had seen this earlier in docker 17.03 as well and this was fixed in docker 18.06 
On our previous AMI's with docker 18.9.9 also, this problem was not happening. Can we get a new AMI for kubernetes 1.15 with docker 18.9.9? Also can this scenario be tested in the AMI release process?

**Comment by harshal-shah** [comment]
We just found that it was our own user-data script that was upgrading the docker version. 
https://github.com/awslabs/amazon-eks-ami/blob/v20200312/eks-worker-al2.json#L17
the default docker version on the AMI does not have this problem. 

having said that, running `yum update -y` can prove dangerous.

~~~
+ yum update -y
Loaded plugins: priorities, update-motd
Existing lock /var/run/yum.pid: another copy is running as pid 3868.
Another app is currently holding the yum lock; waiting for it to exit...
  The other application is: yum
    Memory : 107 M RSS (322 MB VSZ)
    Started: Mon Mar 30 14:29:44 2020 - 00:02 ago
    State  : Running, pid: 3868
Resolving Dependencies
--> Running transaction check
---> Package containerd.x86_64 0:1.2.6-1.amzn2 will be updated
---> Package containerd.x86_64 0:1.3.2-1.amzn2 will be an update
---> Package docker.x86_64 0:18.09.9ce-2.amzn2 will be updated
---> Package docker.x86_64 0:19.03.6ce-1.amzn2 will be an update
---> Package kernel.x86_64 0:4.14.173-137.228.amzn2 will be installed
---> Package runc.x86_64 0:1.0.0-0.1.20190510.git2b18fe1.amzn2 will be updated
---> Package runc.x86_64 0:1.0.0-0.1.20200204.gitdc9208a.amzn2 will be an update
--> Finished Dependency Resolution

Dependencies Resolved

================================================================================
 Package    Arch   Version                              Repository         Size
================================================================================
Installing:
 kernel     x86_64 4.14.173-137.228.amzn2               amzn2-core         20 M
Updating:
 containerd x86_64 1.3.2-1.amzn2                        amzn2extra-docker  24 M
 docker     x86_64 19.03.6ce-1.amzn2                    amzn2extra-docker  37 M
 runc       x86_64 1.0.0-0.1.20200204.gitdc9208a.amzn2  amzn2extra-docker 2.6 M

Transaction Summary
================================================================================
Install  1 Package
Upgrade  3 Packages

~~~
